src/main/java/scrolls/elder/model/Datastore.java

```diff
@@ -0,0 +1,86 @@
+package scrolls.elder.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents the in-memory model of the application data.
+ * The interface through which all other components access application data.
+ * Does not provide mutators, as that is handled by the individual stores.
+ */
+public class Datastore implements ReadOnlyDatastore {
+    private final PersonStore persons;
+    private final LogStore logs;
+
+    /**
+     * Creates an empty Datastore.
+     */
+    public Datastore() {
+        persons = new PersonStore();
+        logs = new LogStore();
+    }
+
+    /**
+     * Creates a Datastore using the data in {@code toBeCopied}
+     */
+    public Datastore(ReadOnlyDatastore toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    //// Datastore accessors
+
+    @Override
+    public ReadOnlyPersonStore getPersonStore() {
+        return persons;
+    }
+
+    public PersonStore getMutablePersonStore() {
+        return persons;
+    }
+
+    @Override
+    public ReadOnlyLogStore getLogStore() {
+        return logs;
+    }
+
+    public LogStore getMutableLogStore() {
+        return logs;
+    }
```

Comment:
I like that you used datastore as a model to represent the application data by storing personstore and logstore!